OpenRA.Mods.Common/Activities/Air/ReturnToBase.cs

```diff
@@ -135,10 +146,40 @@ public override Activity Tick(Actor self)
 				var nearestResupplier = ChooseResupplier(self, false);
 
 				if (nearestResupplier != null)
-					return ActivityUtils.SequenceActivities(self,
-						new Fly(self, Target.FromActor(nearestResupplier), WDist.Zero, aircraft.Info.WaitDistanceFromResupplyBase, targetLineColor: Color.Green),
-						new FlyCircle(self, aircraft.Info.NumberOfTicksToVerifyAvailableAirport),
-						this);
+				{
+					if (aircraft.Info.VTOL)
+					{
+						var distanceFromResupplier = (nearestResupplier.CenterPosition - self.CenterPosition).HorizontalLength;
+						var distanceLength = aircraft.Info.WaitDistanceFromResupplyBase.Length;
+
+						// If no pad is available, move near one and wait
+						if (distanceFromResupplier > distanceLength)
+						{
+							var randomPosition = WVec.FromPDF(self.World.SharedRandom, 2) * distanceLength / 1024;
+							var target = Target.FromPos(nearestResupplier.CenterPosition + randomPosition);
+
+							QueueChild(self, new HeliFly(self, target, WDist.Zero, aircraft.Info.WaitDistanceFromResupplyBase, targetLineColor: Color.Green));
+						}
+
+						return this;
+					}
+					else
+					{
+						QueueChild(self, new Fly(self, Target.FromActor(nearestResupplier), WDist.Zero, aircraft.Info.WaitDistanceFromResupplyBase, targetLineColor: Color.Green));
+						QueueChild(self, new FlyCircle(self, aircraft.Info.NumberOfTicksToVerifyAvailableAirport));
+						return this;
+					}
+				}
+				else if (nearestResupplier == null && aircraft.Info.VTOL && aircraft.Info.LandWhenIdle)
+				{
+					// Using Queue instead of QueueChild here is intentional, as we want VTOLs with LandWhenIdle to land and stay there in this situation
```

Comment:
It might be nicer to have them return to a fallback position (such as the player spawn point or construction yard) instead of landing on the spot as that is usually the exact opposite of what you want when ordering return to base in such a situation. This is probably slightly out of scope though.

Reply:
I agree about the idea, but that's what #15873 (or a follow-up to that) will take care of, so I'd like to leave this as-is in this PR.